client/src/app/appointments/appointments.component.ts

```diff
@@ -40,7 +46,28 @@ export class AppointmentsComponent {
    */
   public onHearingAidsEvaluation() {
     console.log('Hearing Aids Evaluation');
+    this.clearData();
     Utilities.setSessionStorage('appt', 'Hearing Aids Evaluation');
     this.router.navigateByUrl('/ts');
   }
+
+  private clearData(): void {
```

Comment:
It would be nice to detect whether the previous survey data had been submitted, and if not, ask for confirmation first (and if not confirmed, cancel whatever action required this). This may be difficult to implement correctly however.

Reply:
If the previous survey data was submitted, then it should have cleared all appointment data, so the generic-clear-data class will just check if there's any data for it to clear.